File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {

        string result = "", init, temp;
        int n = strs.size();
        init = strs[0];
        for(int i=0; i<init.size(); i++) {
            for(int j=0; j<n; j++) {
                temp = strs[j];
                if(init[i] != temp[i]) {
                    return result;
                }
            }
            result+=init[i];
        }

        return result;
        
    }
};

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        
        if len(strs) == 1:
            return strs[0]
        i = 1
        length = 0
        minL = len(strs[0])
        for s in strs:
            minL = min(minL, len(s))

        prefix = ''
        while i <= minL:
            prefix = strs[0][:i]
            for x in strs:
                if prefix != x[:i]:
                    return prefix[:i-1]
            i += 1

        return prefix


